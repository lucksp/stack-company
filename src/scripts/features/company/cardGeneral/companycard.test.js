import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { CompanyCard } from ".";
import Card from "../../../components/card";
import { Link } from "react-router-dom";

const props = {
  data: {
    "123abc": {
      details: {
        name: "test company",
        address: "123 main",
        revenue: "$100",
        phone: "(720) 201-0040"
      },
      count: 0
    }
  }
};
let wrapper;

describe("<CompanyCard />", () => {
  beforeEach(() => {
    wrapper = shallow(<CompanyCard {...props} />)
      .find(Card)
      .renderProp("renderWith")();
  });
  it("has a snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("has a link", () => {
    expect(wrapper.find(Link).props().to).toEqual({ pathname: "123abc" });
  });
});
