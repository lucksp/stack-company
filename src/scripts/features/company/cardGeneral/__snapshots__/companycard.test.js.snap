// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyCard /> has a snapshot 1`] = `
<Fragment>
  <div
    className="header"
  >
    test company
  </div>
  <div
    className="main"
  >
    <div
      className="company-detail"
    >
      <div
        className="detail title"
      >
        name
      </div>
      <div
        className="detail data"
      >
        test company
      </div>
      <div
        className="detail title"
      >
        address
      </div>
      <div
        className="detail data"
      >
        123 main
      </div>
      <div
        className="detail title"
      >
        revenue
      </div>
      <div
        className="detail data"
      >
        $100
      </div>
      <div
        className="detail title"
      >
        phone
      </div>
      <div
        className="detail data"
      >
        (720) 201-0040
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <Link
      params={
        Object {
          "id": "123abc",
        }
      }
      replace={false}
      to={
        Object {
          "pathname": "123abc",
        }
      }
    >
      Company Overview
    </Link>
  </div>
</Fragment>
`;
